src/amount.cpp

```diff
@@ -9,20 +9,30 @@
 
 const std::string CURRENCY_UNIT = "BTC";
 
-CFeeRate::CFeeRate(const CAmount& nFeePaid, size_t nSize)
+CFeeRate::CFeeRate(const CAmount& nFeePaid, size_t nBytes_)
 {
+    assert(nBytes_ <= uint64_t(std::numeric_limits<int64_t>::max()));
+    int64_t nSize = int64_t(nBytes_);
+
     if (nSize > 0)
-        nSatoshisPerK = nFeePaid*1000/nSize;
+        nSatoshisPerK = nFeePaid * 1000 / nSize;
     else
         nSatoshisPerK = 0;
 }
 
-CAmount CFeeRate::GetFee(size_t nSize) const
+CAmount CFeeRate::GetFee(size_t nBytes_) const
 {
+    assert(nBytes_ <= uint64_t(std::numeric_limits<int64_t>::max()));
+    int64_t nSize = int64_t(nBytes_);
+
     CAmount nFee = nSatoshisPerK * nSize / 1000;
 
-    if (nFee == 0 && nSize != 0 && nSatoshisPerK > 0)
-        nFee = CAmount(1);
+    if (nFee == 0 && nSize != 0) {
```

Comment:
Couldn't have we just removed this special case from here (ie move this check to the callers that need it)?
I know that would be more disruptive, but it will also make increasing the internal precision easier later.